packages/protocol/contracts/L2/TaikoL2.sol

```diff
@@ -139,18 +145,18 @@ contract TaikoL2 is CrossChainOwned, ICrossChainSync {
             revert L2_BASEFEE_MISMATCH();
         }
 
-        // Store the L1's signal root as a signal to the local signal service to
+        // Store the L1's block hash as a signal to the local signal service to
         // allow for multi-hop bridging.
-        ISignalService(resolve("signal_service", false)).sendSignal(l1SignalRoot);
-        emit CrossChainSynced(uint64(block.number), l1Height, l1BlockHash, l1SignalRoot);
+        ISignalService(resolve("signal_service", false)).sendSignal(l1BlockHash);
+        emit CrossChainSynced(uint64(block.number), l1Height, l1BlockHash, parentBlockHeader.stateRoot);
 
         // Update state variables
         l2Hashes[parentId] = blockhash(parentId);
         snippets[l1Height] = ICrossChainSync.Snippet({
             remoteBlockId: l1Height,
             syncedInBlock: uint64(block.number),
             blockHash: l1BlockHash,
-            signalRoot: l1SignalRoot
+            stateRoot: parentBlockHeader.stateRoot
```

Comment:
This is the L2's parentBlock's `stateRoot` not the L1 state root. 🤔 Are you sure this is what we want ?